homeassistant/components/netatmo/sensor.py

```diff
@@ -326,9 +328,10 @@ async def add_public_entities(update=True):
 class NetatmoSensor(NetatmoBase, SensorEntity):
     """Implementation of a Netatmo sensor."""
 
-    def __init__(self, data_handler, data_class_name, module_info, sensor_type):
+    def __init__(self, hass, data_handler, data_class_name, module_info, sensor_type):
         """Initialize the sensor."""
         super().__init__(data_handler)
+        self.hass = hass
```

Comment:
Why is this needed?

Reply:
It's needed because of the debug print on [line 248](https://github.com/home-assistant/core/pull/48261/files/bfbfa9ae34aa39bb474a4f70804dd1cfb013c341..25767ec9623aa9eed7f14543944f2843a55dab5b#diff-9de7b1b1da02e77a56a4aa11f3e134f7d79dad0057325056d35ae721676c545eR248. When the entities are stringified, `Entity.__repr__` calls `Sensor.state` which needs a `hass` object for the unit conversion.

Reply:
I think we should overwrite `__repr__` for the `SensorEntity` class and not include the state. This is going to be a problem in more platforms otherwise. We should not assume that `self.hass` is available when instantiating the entity.

Reply:
OK, that's better.
Fixed in 5074b43